compiler/src/dotty/tools/dotc/config/Feature.scala

```diff
@@ -100,19 +100,20 @@ object Feature:
   private val assumeExperimentalIn = Set("dotty.tools.vulpix.ParallelTesting")
 
   def checkExperimentalFeature(which: String, srcPos: SrcPos = NoSourcePosition)(using Context) =
-    def hasSpecialPermission =
-      new Exception().getStackTrace.exists(elem =>
-        assumeExperimentalIn.exists(elem.getClassName().startsWith(_)))
-    if !(Properties.experimental || hasSpecialPermission)
-       || ctx.settings.YnoExperimental.value
-    then
+    if !isExperimentalEnabled then
       //println(i"${new Exception().getStackTrace.map(_.getClassName).toList}%\n%")
-      report.error(i"Experimental feature$which may only be used with nightly or snapshot version of compiler", srcPos)
+      report.error(i"Experimental $which may only be used with nightly or snapshot version of compiler", srcPos)
 
   /** Check that experimental compiler options are only set for snapshot or nightly compiler versions. */
   def checkExperimentalSettings(using Context): Unit =
     for setting <- ctx.settings.language.value
         if setting.startsWith("experimental.") && setting != "experimental.macros"
-    do checkExperimentalFeature(s" $setting")
+    do checkExperimentalFeature(s"feature $setting")
+
+  def isExperimentalEnabled(using Context): Boolean =
+    def hasSpecialPermission =
+      new Exception().getStackTrace.exists(elem =>
```

Comment:
This isn't new code but we don't need to generate an Exception to get the stacktrace:
```suggestion
      Thread.currentThread.getStackTrace.exists(elem =>
```